pkgs/by-name/sc/screen-pipe/package.nix

```
@@ -0,0 +1,82 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, pkg-config
+, dbus
+, ffmpeg
+, oniguruma
+, openssl
+, sqlite
+, stdenv
+, darwin
+, alsa-lib
+, xorg
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "screen-pipe";
+  version = "0.1.48";
+
+  src = fetchFromGitHub {
+    owner = "louis030195";
+    repo = "screen-pipe";
+    rev = "v${version}";
+    hash = "sha256-rWKRCqWFuPO84C52mMrrS4euD6XdJU8kqZsAz28+vWE=";
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "cpal-0.15.3" = "sha256-eKn3tS5QuqbMTwnRAEybvbPZOiKiid7ghGztAmrs9fw=";
+      "rusty-tesseract-1.1.10" = "sha256-XT74zGn+DetEBUujHm4Soe2iorQcIoUeZbscTv+64hw=";
+    };
+  };
+
+  postPatch = ''
+    ln -s ${./Cargo.lock} Cargo.lock
+  '';
+
+  nativeBuildInputs = [
+    pkg-config
+    rustPlatform.bindgenHook
+  ];
+
+  buildInputs = [
+    dbus
+    ffmpeg
+    oniguruma
+    openssl
+    sqlite
+  ] ++ lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk_12_3.frameworks.CoreAudio
+    darwin.apple_sdk_12_3.frameworks.AudioUnit
+    darwin.apple_sdk_12_3.frameworks.CoreFoundation
+    darwin.apple_sdk_12_3.frameworks.CoreGraphics
+    darwin.apple_sdk_12_3.frameworks.CoreMedia
+    darwin.apple_sdk_12_3.frameworks.IOKit
+    darwin.apple_sdk_12_3.frameworks.Metal
+    darwin.apple_sdk_12_3.frameworks.MetalPerformanceShaders
+    darwin.apple_sdk_12_3.frameworks.Security
+    darwin.apple_sdk_12_3.frameworks.ScreenCaptureKit
+    darwin.apple_sdk_12_3.frameworks.SystemConfiguration
+  ] ++ lib.optionals stdenv.isLinux [
```

Using `with` here would make for more readable and maintainable darwin `buildInputs`
```suggestion
  ] ++ lib.optionals stdenv.isDarwin (with darwin.apple_sdk_12_3.frameworks; [
    CoreAudio
    AudioUnit
    CoreFoundation
    CoreGraphics
    CoreMedia
    IOKit
    Metal
    MetalPerformanceShaders
    Security
    ScreenCaptureKit
    SystemConfiguration
  ]) ++ lib.optionals stdenv.isLinux [
```